From f16cb5186e0a09e5186dde3855de650d8274228a Mon Sep 17 00:00:00 2001
From: Serge <mail@olegnax.com>
Date: Mon, 30 Jul 2012 19:33:21 +0300
Subject: [PATCH 05/10] Add an option to the Products panel in the Theme
 Settings page to allow merchant to optionally display
 the product SKU on the product detail page

---
 .../local/Olegnax/Celebritysettings/etc/config.xml |    1 +
 .../local/Olegnax/Celebritysettings/etc/system.xml |   10 +++
 .../catalog/product/view/type/default.phtml        |   39 ++++++++
 .../catalog/product/view/type/grouped.phtml        |   94 ++++++++++++++++++++
 skin/frontend/default/celebrity/css/local.css      |    2 +
 5 files changed, 146 insertions(+)
 create mode 100644 app/design/frontend/default/celebrity/template/catalog/product/view/type/default.phtml
 create mode 100644 app/design/frontend/default/celebrity/template/catalog/product/view/type/grouped.phtml

diff --git a/app/code/local/Olegnax/Celebritysettings/etc/config.xml b/app/code/local/Olegnax/Celebritysettings/etc/config.xml
index 3b4fdd4..328887e 100644
--- a/app/code/local/Olegnax/Celebritysettings/etc/config.xml
+++ b/app/code/local/Olegnax/Celebritysettings/etc/config.xml
@@ -170,6 +170,7 @@
 				<new_label_position>top-left</new_label_position>
 				<sale_label>1</sale_label>
 				<sale_label_position>top-right</sale_label_position>
+                <show_sku>1</show_sku>
 			</celebritysettings_labels>
 		</celebritysettings>
 	</default>
diff --git a/app/code/local/Olegnax/Celebritysettings/etc/system.xml b/app/code/local/Olegnax/Celebritysettings/etc/system.xml
index 63dfe3e..ca2d68f 100644
--- a/app/code/local/Olegnax/Celebritysettings/etc/system.xml
+++ b/app/code/local/Olegnax/Celebritysettings/etc/system.xml
@@ -282,6 +282,16 @@
                             <show_in_website>1</show_in_website>
                             <show_in_store>1</show_in_store>
                         </sale_label_position>
+                        <show_sku translate="label">
+                            <label>Show Item SKU</label>
+                            <comment>Show SKU on product details page</comment>
+                            <frontend_type>select</frontend_type>
+                            <source_model>adminhtml/system_config_source_yesno</source_model>
+                            <sort_order>50</sort_order>
+                            <show_in_default>1</show_in_default>
+                            <show_in_website>1</show_in_website>
+                            <show_in_store>1</show_in_store>
+                        </show_sku>
                     </fields>
                 </celebritysettings_labels>
 			</groups>
diff --git a/app/design/frontend/default/celebrity/template/catalog/product/view/type/default.phtml b/app/design/frontend/default/celebrity/template/catalog/product/view/type/default.phtml
new file mode 100644
index 0000000..2c3ae45
--- /dev/null
+++ b/app/design/frontend/default/celebrity/template/catalog/product/view/type/default.phtml
@@ -0,0 +1,39 @@
+<?php
+/**
+ * Magento
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License (AFL 3.0)
+ * that is bundled with this package in the file LICENSE_AFL.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/afl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@magentocommerce.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade Magento to newer
+ * versions in the future. If you wish to customize Magento for your
+ * needs please refer to http://www.magentocommerce.com for more information.
+ *
+ * @category    design
+ * @package     base_default
+ * @copyright   Copyright (c) 2012 Magento Inc. (http://www.magentocommerce.com)
+ * @license     http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+ */
+?>
+<?php /* @var $this Mage_Catalog_Block_Product_View_Abstract */?>
+<?php $_product = $this->getProduct() ?>
+
+<?php if ($_product->isAvailable()): ?>
+    <p class="availability in-stock"><?php echo $this->__('Availability:') ?> <span><?php echo $this->__('In stock') ?></span></p>
+<?php else: ?>
+    <p class="availability out-of-stock"><?php echo $this->__('Availability:') ?> <span><?php echo $this->__('Out of stock') ?></span></p>
+<?php endif; ?>
+<?php if (Mage::getStoreConfig('celebritysettings/celebritysettings_labels/show_sku')): ?>
+    <p class="sku"><?php echo $this->__('Item #:') ?><span><?php echo $_product->getSku() ?></span></p>
+<?php endif; ?>
+<?php echo $this->getChildHtml('product_type_data_extra') ?>
+<?php echo $this->getPriceHtml($_product) ?>
diff --git a/app/design/frontend/default/celebrity/template/catalog/product/view/type/grouped.phtml b/app/design/frontend/default/celebrity/template/catalog/product/view/type/grouped.phtml
new file mode 100644
index 0000000..ad4d642
--- /dev/null
+++ b/app/design/frontend/default/celebrity/template/catalog/product/view/type/grouped.phtml
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Magento
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License (AFL 3.0)
+ * that is bundled with this package in the file LICENSE_AFL.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/afl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@magentocommerce.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade Magento to newer
+ * versions in the future. If you wish to customize Magento for your
+ * needs please refer to http://www.magentocommerce.com for more information.
+ *
+ * @category    design
+ * @package     base_default
+ * @copyright   Copyright (c) 2012 Magento Inc. (http://www.magentocommerce.com)
+ * @license     http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+ */
+
+/**
+ * Grouped product data template
+ *
+ * @see Mage_Catalog_Block_Product_View_Media
+ * @see Mage_Catalog_Block_Product_View_Type_Grouped
+ */
+?>
+<?php $this->setPreconfiguredValue(); ?>
+<?php $_product = $this->getProduct(); ?>
+<?php $_associatedProducts = $this->getAssociatedProducts(); ?>
+<?php $_hasAssociatedProducts = count($_associatedProducts) > 0; ?>
+<?php if ($_product->isAvailable() && $_hasAssociatedProducts): ?>
+    <p class="availability in-stock"><?php echo $this->__('Availability:') ?> <span><?php echo $this->__('In stock') ?></span></p>
+<?php else: ?>
+    <p class="availability out-of-stock"><?php echo $this->__('Availability:') ?> <span><?php echo $this->__('Out of stock') ?></span></p>
+<?php endif; ?>
+<?php if (Mage::getStoreConfig('celebritysettings/celebritysettings_labels/show_sku')): ?>
+<p class="sku"><?php echo $this->__('Item #:') ?> <span><?php echo $_product->getSku() ?></span></p>
+<?php endif; ?>
+<?php echo $this->getChildHtml('product_type_data_extra') ?>
+<table class="data-table grouped-items-table" id="super-product-table">
+    <col />
+    <col />
+    <col width="1" />
+    <thead>
+        <tr>
+            <th><?php echo $this->__('Product Name') ?></th>
+            <?php if ($this->getCanShowProductPrice($_product)): ?>
+            <th class="a-right"><?php echo $this->__('Price') ?></th>
+            <?php endif; ?>
+            <?php if ($_product->isSaleable()): ?>
+            <th class="a-center"><?php echo $this->__('Qty') ?></th>
+            <?php endif; ?>
+        </tr>
+    </thead>
+    <tbody>
+    <?php if ($_hasAssociatedProducts): ?>
+    <?php foreach ($_associatedProducts as $_item): ?>
+        <?php $_finalPriceInclTax = $this->helper('tax')->getPrice($_item, $_item->getFinalPrice(), true) ?>
+        <tr>
+            <td><?php echo $this->htmlEscape($_item->getName()) ?></td>
+            <?php if ($this->getCanShowProductPrice($_product)): ?>
+            <td class="a-right">
+                <?php if ($this->getCanShowProductPrice($_item)): ?>
+                <?php echo $this->getPriceHtml($_item, true) ?>
+                <?php echo $this->getTierPriceHtml($_item) ?>
+                <?php endif; ?>
+            </td>
+            <?php endif; ?>
+            <?php if ($_product->isSaleable()): ?>
+            <td class="a-center">
+            <?php if ($_item->isSaleable()) : ?>
+                <input type="text" name="super_group[<?php echo $_item->getId() ?>]" maxlength="12" value="<?php echo $_item->getQty()*1 ?>" title="<?php echo $this->__('Qty') ?>" class="input-text qty" />
+            <?php else: ?>
+                <p class="availability out-of-stock"><span><?php echo $this->__('Out of stock') ?></span></p>
+            <?php endif; ?>
+            </td>
+            <?php endif; ?>
+        </tr>
+    <?php endforeach; ?>
+    <?php else: ?>
+       <tr>
+           <td colspan="<?php if ($_product->isSaleable()): ?>4<?php else : ?>3<?php endif; ?>"><?php echo $this->__('No options of this product are available.') ?></td>
+       </tr>
+    <?php endif; ?>
+    </tbody>
+</table>
+<script type="text/javascript">decorateTable('super-product-table')</script>
diff --git a/skin/frontend/default/celebrity/css/local.css b/skin/frontend/default/celebrity/css/local.css
index 2418fc6..35badd8 100644
--- a/skin/frontend/default/celebrity/css/local.css
+++ b/skin/frontend/default/celebrity/css/local.css
@@ -910,6 +910,8 @@ nav {float:left;}
 .product-view h2 {font-size:20px; color:#4f4f4f;}
 .product-view .product-name { margin-bottom:0px; }
 .product-view .availability { font-size:11px; margin: 14px 0; }
+.product-view .sku { font-size:11px; margin:-14px 0 14px; }
+.product-view .sku span { font-weight:700}
 .product-view .price-box { font-size:18px; color:#252525 }
 .product-view .product-essential {}
 .product-view .short-description { margin:18px 0; padding:18px 0; border-bottom:1px solid #e7e7e7; border-top:1px solid #e7e7e7; }
-- 
1.7.10.msysgit.1

